subcommands/keys/ca_revoke_device_ca.go

```diff
@@ -0,0 +1,224 @@
+- Existing devices with client certificates issued by this device CA can no longer connect to your FoundriesFactory.
+
+You may later re-add a revoked device CA using "keys ca update", if you still keep its certificate stored somewhere.
+Once you do it, devices with client certificates issued by this device CA may connect to your FoundriesFactory again.`,
+		Annotations: map[string]string{crlCmdAnnotation: crlCmdRevokeDeviceCa},
+	}
+	caCmd.AddCommand(revokeCmd)
+	addRevokeCmdFlags(revokeCmd, "revoke")
+
+	disableCmd := &cobra.Command{
+		Use:   "disable-device-ca <PKI Directory>",
+		Short: "Disable device CA devices with client certificates issued by it can no longer be registered",
+		Run:   doRevokeDeviceCa,
+		Args:  cobra.ExactArgs(1),
+		Long: `Disable device CA devices with client certificates issued by it can no longer be registered.
+
+When the online or local device CA is disabled:
+- It is no longer possible to register new devices with client certificates issued by this device CA.
+- Existing devices with client certificates issued by this device CA can continue to connect and use FoundriesFactory.
+
+Usually, when the device CA is compromised, a user would:
+1. Immediately disable a given device CA using "fioctl keys ca disable-device-ca <PKI Directory> --serial <CA Serial>".
+2. Inspect their devices with client certificates issued by that device CA, and remove any compromised devices.
+3. Create a new device CA using "fioctl keys ca add-device-ca <PKI Directory> --online-ca|--local-ca".
+4. Rotate a client certificate of all legitimate devices from step 2 to the certificate issued by a new device CA.
+5. Revoke a given device CA using "fioctl keys ca revoke-device-ca <PKI Directory> --serial <CA Serial>".`,
+		Annotations: map[string]string{crlCmdAnnotation: crlCmdDisableDeviceCa},
+	}
+	caCmd.AddCommand(disableCmd)
+	addRevokeCmdFlags(disableCmd, "disable")
+}
+
+func addRevokeCmdFlags(cmd *cobra.Command, op string) {
+	cmd.Flags().BoolP("dry-run", "", false,
+		"Do not "+op+" the certificate, but instead show a generated CRL that will be uploaded to the server.")
+	cmd.Flags().BoolP("pretty", "", false,
+		"Can be used with dry-run to show the generated CRL in a pretty format.")
+	cmd.Flags().StringArrayP("ca-file", "", nil,
+		"A file name of the device CA to "+op+". Can be used several times to "+op+" several device CAs")
+	cmd.Flags().StringArrayP("ca-serial", "", nil,
+		"A serial number (base 10) of the device CA to "+op+". Can be used several times to "+op+" several device CAs")
+	_ = cmd.MarkFlagFilename("ca-file")
+	// HSM variables defined in ca_create.go
+	cmd.Flags().StringVarP(&hsmModule, "hsm-module", "", "", "Load a root CA key from a PKCS#11 compatible HSM using this module")
+	cmd.Flags().StringVarP(&hsmPin, "hsm-pin", "", "", "The PKCS#11 PIN to log into the HSM")
+	cmd.Flags().StringVarP(&hsmTokenLabel, "hsm-token-label", "", "", "The label of the HSM token containing the root CA key")
```

Comment:
Does fioctl also create those? Am I right? Can we add the file name we created for those cases?

Reply:
Fioctl is a user here. There are no file names on the HSM.

Reply:
I see we just received this input and we never generate those files.
If so fair enough.

Reply:
Yes, exactly.

IIRC I added to the README.md how to integrate with the SoftHSM some time ago.
So, it is kind of _loosly_ self-documented; also there should be something about it in docs.
But, we're not going to copy-paste that in every single command; that'd be onerous to maintain across the product.